### ISSUE TITLE
How to use this module?

### ISSUE THREAD

**Issue by cytopia** [hashicorp/terraform-aws-vault#103]
# Usage confusion

I am a little bit confused about the usage of this module.
Should I use it like so:
```
module "terraform_aws_vault" {
  source = "github.com/hashicorp/terraform-aws-vault?ref=v0.10.3"

  ...
}
```
But then it won't allow me to customize security groups, however in your `main.tf` [it clearly states](https://github.com/hashicorp/terraform-aws-vault/blob/master/main.tf#L187):
>   ```
> # To make testing easier, we allow Consul and SSH requests from any IP address
> # here but in a production deployment, we strongly recommend you limit this to the
> # IP address ranges of known, trusted servers inside your VPC.
> ```
---

Or am I supposed to copy `main.tf`, `variables.tf` and `outputs.tf` locally and create my own module based on the submodules you provide in the [modules/](https://github.com/hashicorp/terraform-aws-vault/tree/master/modules) directory?

Something like this:
```
module "vault_cluster" {
  source = "github.com/hashicorp/terraform-aws-vault//modules/vault-cluster?ref=v0.0.1"
  ...
}
module "consul_iam_policies_servers" {
  source = "github.com/hashicorp/terraform-aws-consul//modules/consul-iam-policies?ref=v0.4.0"
  ...
}
module "security_group_rules" {
  source = "github.com/hashicorp/terraform-aws-consul.git//modules/consul-client-security-group-rules?ref=v0.4.0"
  ...
}
module "vault_elb" {
  source = "github.com/hashicorp/terraform-aws-vault//modules/vault-elb?ref=v0.0.1"
  ...
}
module "consul_cluster" {
  source = "github.com/hashicorp/terraform-aws-consul//modules/consul-cluster?ref=v0.4.0"
  ...
}
...
```

**Edit**

I was simply looking for something like this as a general guidance for how to actually use this module: https://github.com/terraform-aws-modules/terraform-aws-vpc#usage

Thanks

**Comment by Etiene** [comment]
> Or am I supposed to copy main.tf, variables.tf and outputs.tf locally and create my own module based on the submodules you provide in the modules/ directory?

Yes. The module you want to use is in fact the `vault_cluster` module located at `/modules`. The main.tf at this root dir is just one example on how to use it. There are other examples located at `/examples`
